src/pools/cluster/dynamic.ts

```diff
@@ -58,11 +51,11 @@ export class DynamicClusterPool<
         return super.chooseWorker()
       }
       // all workers are busy create a new worker
-      const worker = this.newWorker()
-      worker.on('message', (message: { kill?: number }) => {
+      const worker = this.internalNewWorker()
+      worker.on('message', (message: MessageValue<Data>) => {
         if (message.kill) {
-          worker.send({ kill: 1 })
-          worker.kill()
+          this.sendToWorker(worker, { kill: 1 })
+          void this.destroyWorker(worker)
           // clean workers from data structures
           const workerIndex = this.workers.indexOf(worker)
           this.workers.splice(workerIndex, 1)
```

Comment:
IMO this cleanup should be part of the destroy code (similar to the `internalNewWorker`), because it operates on the fields of the superclass.

Reply:
@jerome-benoit Seems to already work on that => https://github.com/pioardi/poolifier/pull/124